### PR TITLE
Improve OAuth handling

### PR DESCRIPTION
**Pull request by rullzer**

### CODE REVIEW

File: apps/oauth2/lib/Controller/LoginRedirectorController.php [hunk 1]
```diff
@@ -61,11 +61,20 @@ public function __construct($appName,
 	 *
 	 * @param string $client_id
 	 * @param string $state
+	 * @param string $response_type
 	 * @return RedirectResponse
 	 */
 	public function authorize($client_id,
-							  $state) {
+							  $state,
+							  $response_type) {
```
Comment (blizzz) [inline apps/oauth2/lib/Controller/LoginRedirectorController.php]: camelCase?

File: apps/oauth2/lib/Controller/OauthApiController.php [hunk 2]
```diff
@@ -21,67 +21,151 @@
...
+	 * @param string $grant_type
 	 * @param string $code
+	 * @param string $refresh_token
+	 * @param string $client_id
+	 * @param string $client_secret
 	 * @return JSONResponse
 	 */
-	public function getToken($code) {
-		$accessToken = $this->accessTokenMapper->getByCode($code);
+	public function getToken($grant_type, $code, $refresh_token, $client_id, $client_secret) {
```
Comment (blizzz) [inline apps/oauth2/lib/Controller/OauthApiController.php]: cases

**Review by blizzz — COMMENTED** [review]
cannot say much about it… but at least some bikeshedding

File: apps/oauth2/lib/Controller/LoginRedirectorController.php (hunk 1 above)
Comment (rullzer) [reply apps/oauth2/lib/Controller/LoginRedirectorController.php]: No I can't change those as they are post variables and defined in the OAuth spec

File: apps/oauth2/lib/Controller/OauthApiController.php (hunk 2 above)
Comment (rullzer) [reply apps/oauth2/lib/Controller/OauthApiController.php]: No I can't change those as they are post variables and defined in the OAuth spec

File: apps/oauth2/lib/Controller/LoginRedirectorController.php (hunk 1 above)
Comment (blizzz) [reply apps/oauth2/lib/Controller/LoginRedirectorController.php]: okay then

File: apps/oauth2/lib/Controller/OauthApiController.php (hunk 2 above)
Comment (blizzz) [reply apps/oauth2/lib/Controller/OauthApiController.php]: okay then

File: apps/oauth2/lib/Migration/SetTokenExpiration.php [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+		$tokens = [];
+		while($row = $cursor->fetch()) {
+			$tokens[] = AccessToken::fromRow($row);
+		}
+		$cursor->closeCursor();
+
+		// Set the 3600 second timeout on all tokens
+		foreach ($tokens as $token) {
+			try {
+				$appToken = $this->tokenProvider->getTokenById($token->getTokenId());
```
Comment (blizzz) [inline apps/oauth2/lib/Migration/SetTokenExpiration.php]: would it have a better memory footprint if we deal with that inside the while-loop above?

Comment (rullzer) [reply apps/oauth2/lib/Migration/SetTokenExpiration.php]: yes fair enough let me fix that...

Comment (rullzer) [reply apps/oauth2/lib/Migration/SetTokenExpiration.php]: done

File: apps/oauth2/lib/Controller/OauthApiController.php [hunk 4]
```diff
@@ -21,67 +21,151 @@
...
+		try {
+			$client = $this->clientMapper->getByUid($accessToken->getClientId());
+		} catch (ClientNotFoundException $e) {
+			return new JSONResponse([
+				'error' => 'invalid_request',
+			], Http::STATUS_BAD_REQUEST);
+		}
+
+		// The client id and secret must match. Else we don't provide an access token!
+		if ($client->getClientIdentifier() !== $client_id || $client->getSecret() !== $client_secret) {
```
Comment (blizzz) [inline apps/oauth2/lib/Controller/OauthApiController.php]: with grant_type = 'refresh_token' client_id and and secret are not provided. Unless I oversaw a thing?

**Review by blizzz — APPROVED** [review]
looks good to me

File: apps/oauth2/lib/Migration/SetTokenExpiration.php [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+	}
+
+	public function run(IOutput $output) {
+		$qb = $this->connection->getQueryBuilder();
+		$qb->select('*')
+			->from('oauth2_access_tokens');
+
+		$cursor = $qb->execute();
+
+		$tokens = [];
```
Comment (ChristophWurst) [inline apps/oauth2/lib/Migration/SetTokenExpiration.php]: unused

**Review by ChristophWurst — DISMISSED** [review]
Code looks good

Found one unused variable

**Review by ChristophWurst — APPROVED** [review]

**Pull request merged**